functors_task_1/student/src/impl.cpp

```diff
@@ -1,2 +1,29 @@
 #include "impl.h"
 
+BeerBrand BeerOrganizer::operator()() {
+    if (static_cast<size_t>(BeerBrand::Max) == currentIndex)
+        currentIndex = 1;
+
+    return static_cast<BeerBrand>(currentIndex++);
+}
+
+bool isExpensiveBeer(const BeerBrand& brand)
+{
+    std::vector<BeerBrand> expensive{BeerBrand::HoeGaarden, BeerBrand::Corona};
```

Comment:
1. Why do you need dynamic container?
2. Why not the plain logical expression?

Reply:
The purpose was to reuse existing find algorithm, but in this case it seems to be an overhead. Changed to switch - case.